### PR TITLE
fix: Regular error showing Total timeout of API google.logging.v2.LoggingServiceV2 exceeded 600000 milliseconds

### PR DESCRIPTION
**Pull request by losalex**
This is a proposed fix to accept a global callback for `Log.delete` and `Log.write` methods through `LogOptions`

Fixes #[<1185>](https://github.com/googleapis/nodejs-logging/issues/1185) 🦕

### CODE REVIEW

File: src/log.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export interface LogOptions {
   removeCircular?: boolean;
   maxEntrySize?: number; // see: https://cloud.google.com/logging/quotas
   jsonFieldsToTruncate?: string[];
+  defaultWriteDeleteCallback?: ApiResponseCallback;
```
Comment (summer-ji-eng) [inline src/log.ts:63]: Is this global callback only for WriteDelete? I thought you want to have a global callback for all methods in Log class in manual layer.

Comment (losalex) [reply src/log.ts:63]: It is for `write()` and `delete()` APIs which are the only one's used to update logs. Just wanted to distinguish it to make sure we do not confuse developers

**Review by summer-ji-eng — APPROVED** [review]

File: src/log.ts (hunk 1 above)
Comment (minherz) [reply src/log.ts:63]: a small correction:`delete()` does not update logs, it deletes all log entries for a provided log name as a result of the delete log operation while `write()` creates one or more log entries.

i am not sure about the "global" scope. the scope is bound by the `Log` instance, isn't it?

**Review by minherz — COMMENTED** [review]
From the issue (#1185) it is not clear what developers except to get. The complains sound like they do not want an application to crash in a case the retries fail to ingest the log.
In this respect, the fix does not resolve the problem if the callback is not provided.

File: src/log.ts (hunk 1 above)
Comment (losalex) [reply src/log.ts:63]: > 

Based on user's feedback in a bug, the suggestion was: "I assume a lot of users will want to set up some sort of global error catching for this particular error."
Do you have specific proposal in mind how to fix this?

**Comment by losalex** [comment]
> From the issue (#1185) it is not clear what developers except to get. The complains sound like they do not want an application to crash in a case the retries fail to ingest the log. In this respect, the fix does not resolve the problem if the callback is not provided.

Based on user's feedback in a bug, the suggestion was: "I assume a lot of users will want to set up some sort of global error catching for this particular error."
Do you have specific proposal in mind how to fix this?

**Comment by minherz** [comment]
The quoted suggestion comes from one out of four people who describe scenarios that generally caused by failure in retrying mechanism due to transient connectivity error or because the ran their application in the not connected environment.

[One of the cases](https://github.com/googleapis/nodejs-logging/issues/1185#issuecomment-982324947) resulted in printing the error to STDOUT where it was catched by a logging agent and eventually reported to Cloud Logging backend.

In my opinion the problem is when this error crashes the application. It is a good idea to provide a way to define the error callback. I suggest to add a default error handling if nothing is provided, so the application would not crash.

**Comment by losalex** [comment]
> The quoted suggestion comes from one out of four people who describe scenarios that generally caused by failure in retrying mechanism due to transient connectivity error or because the ran their application in the not connected environment.
> 
> [One of the cases](https://github.com/googleapis/nodejs-logging/issues/1185#issuecomment-982324947) resulted in printing the error to STDOUT where it was catched by a logging agent and eventually reported to Cloud Logging backend.
> 
> In my opinion the problem is when this error crashes the application. It is a good idea to provide a way to define the error callback. I suggest to add a default error handling if nothing is provided, so the application would not crash.

I believe we cannot provide default callback since client may call `log.write` with `await` and put the call inside `try/catch` - this way default callback from code will "swallow" any thrown error, thus catch becomes to be redundant. Worth mentioning that it is eventually will violate our contract with users who does `await` with `try/catch`.

File: .readme-partials.yml [hunk 2]
```diff
@@ -71,3 +71,58 @@ body: |-
...
+      try {
+        await log.write(entry);
+      } catch (err) {
+        console.log('Error is: ' + err);
+      }
+  ```
+
+  However, awaiting for every `log.write` or `log.delete` calls may introduce delays which could be avoided by 
+  simply adding a callback like in example below - this way the log entry can be queued for processing and code
+  execution will continue without further delays, so callback will be called once operation is complete:
```
Comment (keenan-devrel) [inline .readme-partials.yml:92]: in the example below. This way the log entry can be queued for processing and code
  execution will continue without further delays.  The callback will be called once the operation is complete:

Comment (losalex) [reply .readme-partials.yml:92]: Thanks!

**Pull request merged**